lib/takeaway.rb

```diff
@@ -0,0 +1,79 @@
+require "csv"
+require_relative 'dish_list'
+require_relative 'dish'
+require_relative 'order'
+require_relative 'sms'
+
+class Takeaway
+
+  attr_reader :dish_list, :order
+
+  def initialize( args = {dish_list: DishList.new(Dish), order: Order.new} )
+    @dish_list     = args[:dish_list]
+    @order         = args[:order]
+    @placed_orders = []
+  end
+
+  def load_dishes(filename='menu.csv')
+    if File.exist?(filename)
+      CSV.foreach(filename) { |data| dish_list.add_dish(data) }
+    else
+      fail "File '#{filename}' not found!"
+    end
+  end
+
+  def menu
+    puts dish_list
+  end
+
+  def add_dish(id, qty=1)
+    fail_if_wrong_id(id)
+    fail_if_wrong_qty(qty)
+    dish = dish_list.select_dish(id)
+    order.add_item(dish, qty)
+  end
+
+  def remove_dish(id, qty=1)
+    fail_if_wrong_id(id)
+    fail_if_wrong_qty(qty)
+    dish = dish_list.select_dish(id)
+    order.remove_item(dish, qty)
+  end
+
+  def order_total
+    order.total
+  end
+
+  def place_order
```

Comment:
The name of this method suggests that I should pass the order as a parameter (and indeed you should).  The taking of the order is handled by `Order` and `Menu`.  The takeaway should process the finalized order.